File: test/fixtures/elixir/put_basic_auth_json_data.ex
response = HTTPoison.put!(
  "http://localhost:28139/test/_security",
  "{\"admins\":{\"names\":[], \"roles\":[]}, \"readers\":{\"names\":[\"joe\"],\"roles\":[]}}",
  [
    {"Content-Type", "application/x-www-form-urlencoded"}
  ],
  [hackney: [basic_auth: {"admin", "123"}]]
)

File: test/fixtures/csharp/put_basic_auth_json_data.cs
using System.Net.Http;
using System.Net.Http.Headers;

HttpClient client = new HttpClient();

HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Put, "http://localhost:28139/test/_security");

request.Headers.Add("Authorization", "Basic " + Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes("admin:123")));

request.Content = new StringContent("{\"admins\":{\"names\":[], \"roles\":[]}, \"readers\":{\"names\":[\"joe\"],\"roles\":[]}}");
request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/x-www-form-urlencoded");

HttpResponseMessage response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
string responseBody = await response.Content.ReadAsStringAsync();
